js/AngularApp.js

```diff
@@ -0,0 +1,51 @@
+var todoApp = angular.module('ToDoApp', []);
+
+todoApp.controller('ToDoAppController', function($scope) {
+  
+  $scope.items = [];
+
+  $scope.addNewTask = function() {
+    var newTask = new Task($scope.newTask);
+    $scope.items.push(newTask);
+  };
+
+  $scope.deleteTask = function(item) {
+    var index = $scope.items.indexOf(item);
+    $scope.items.splice(index, 1);
+  };
+
+  $scope.deleteAllTasks = function() {
+    $scope.items = [];
+  };
+  
+  $scope.changeCompletionStatus = function(item) {
+    (item.isCompleted === true) ? item.isCompleted = false : item.isCompleted = true;
+  };
+  
+  $scope.remainingTasks = function() {
+    var remainingTasks = [];
+    $scope.items.filter(function(task) {
+      if(task.isCompleted === false) {
+        remainingTasks.push(task);
+      }
+    });
+    return remainingTasks.length + " task" + ((remainingTasks.length > 1 || remainingTasks.length === 0) ? "s" : "") + " remaining";
+  };
+
+  $scope.showTasks = function  (task) {
+    if($scope.show === 'All') {
+      return true;
+    } else if (task.isCompleted && $scope.show === "Complete") {
+      return true;
+    } else if (!task.isCompleted && $scope.show === "Active") {
+      return true;
+    } else {
+      return false;
+    }
+  };
```

Comment:
I know you've already mentioned a need for refactoring but I'm sure someone will pick up on using if/else statements here, not that I'm certain I can offer a better way around it right now.